File: code/ch02/2.1.1.twoSum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    int n = nums.size();
    sort(nums.begin(), nums.end());
    int l = 0;
    int r = n - 1;
    while (l < r) {
      if (nums[l] + nums[r] < target) {
        l += 1;
      } else if (nums[l] + nums[r] > target) {
        r -= 1;
      } else {
        return {l, r};
      }
    }
    return {};
  }
};

File: code/ch02/2.1.1.twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        nums.sort()
        l = 0
        r = n - 1
        while l < r:
            if nums[l] + nums[r] < target:
                l += 1
            elif nums[l] + nums[r] > target:
                r -= 1
            else:
                return [l, r]
        return []
